### PR TITLE
Add stderror! method, distinguish coef and fixef

### PR DESCRIPTION
**Pull request by dmbates**
The docs should be updated to indicate that `coef` always refers to the full coefficient vector, as does `stderror` and `fixef` refers to the truncated coefficient vector.  Furthermore this extends to associated functions like `coefnames` and `fixefnames`.  However, I think it is better to do this in the version from #283

### CODE REVIEW

**Comment by dmbates** [comment]
I cancelled the windows-2019 check because it had been running for nearly 40 minutes.

File: src/linearmixedmodel.jl [hunk 1]
```diff
@@ -159,15 +159,21 @@ fit(
...
 )
 
-StatsBase.coef(m::MixedModel) = fixef(m, false)
+function StatsBase.coef(m::LinearMixedModel{T}) where {T}
+    piv = fetrm(m).piv
+    invpermute!(fixef!(similar(piv, T), m), piv)
+end
 
-βs(m::LinearMixedModel) = NamedTuple{(Symbol.(fixefnames(m))...,)}((fixef(m)...,))
+βs(m::LinearMixedModel) = NamedTuple{(Symbol.(coefnames(m))...,)}(coef(m))
```
Comment (palday) [inline src/linearmixedmodel.jl]: `GeneralizedLinearMixedModel` has a property `beta`: not the quite the same as the accessor here, but should standardize? And/or do a `MixedModel` method instead -- `coefnames` and `coef` are defined for both GLMM and LMM.

File: src/linearmixedmodel.jl [hunk 2]
```diff
@@ -285,13 +291,30 @@ function StatsBase.dof_residual(m::LinearMixedModel)::Int
...
+    k = feind(m)
+    LowerTriangular(m.L.blocks[k, k])
+end
+
+"""
+    fetrm(m::LinearMixedModel)
+
+Return the fixed-effects term from `m.allterms`
+"""
+fetrm(m) = m.allterms[feind(m)]
```
Comment (palday) [inline src/linearmixedmodel.jl]: I like having these helpers around.

File: src/linearmixedmodel.jl [hunk 3]
```diff
@@ -356,53 +380,44 @@ StatsBase.fitted(m::LinearMixedModel{T}) where {T} = fitted!(Vector{T}(undef, no
 """
     fixef!(v::Vector{T}, m::LinearMixedModel{T})
 
-Overwrite `v` with the pivoted and, possibly, truncated fixed-effects coefficients of model `m`
-"""
-fixef!(v::AbstractVector{T}, m::LinearMixedModel{T}) where {T} =
-    ldiv!(feL(m)', copyto!(v, m.L.blocks[end, end-1]))
+Overwrite `v` with the pivoted fixed-effects coefficients of model `m`
 
+For full rank models the length of `v` must be the rank of `X`.  For rank-deficient models
```
Comment (palday) [inline src/linearmixedmodel.jl]: "full-rank" for consistency with "rank-deficient"

File: src/linearmixedmodel.jl [hunk 4]
```diff
@@ -880,7 +919,7 @@ function StatsBase.vcov(m::LinearMixedModel{T}; corr=false) where {T}
         vv = covmat[iperm, iperm]
     end
 
-    corr ?  StatsBase.cov2cor!(vv,stderror(m)) : vv
+    corr ?  StatsBase.cov2cor!(vv, stderror(m)) : vv
```
Comment (palday) [inline src/linearmixedmodel.jl]: Did you want to redefine `stderror` to redirect to `stderror!` plus a pre-allocated scratch like `fixef` and `fixef!`?

**Comment by palday** [comment]
> I cancelled the windows-2019 check because it had been running for nearly 40 minutes.

It's Tier 2 anyway :)

File: src/linearmixedmodel.jl (hunk 1 above)
Comment (dmbates) [reply src/linearmixedmodel.jl]: I think it is best to keep separate methods for `LinearMixedModel` and `GeneralizedLinearMixedModel`.  When beta is part of the "baba yaga" optimization you can't guarantee that the calculation using only the LMM field will produce the correct answer.  This is related to the distinction between `fast=true` and `fast=false`.

File: src/linearmixedmodel.jl (hunk 2 above)
Comment (dmbates) [reply src/linearmixedmodel.jl]: :+1:

File: src/linearmixedmodel.jl (hunk 3 above)
Comment (dmbates) [reply src/linearmixedmodel.jl]: Thanks for catching that.  I will fix it.

File: src/linearmixedmodel.jl (hunk 4 above)
Comment (dmbates) [reply src/linearmixedmodel.jl]: I suppose now that I have written `stderror!` we should use it.  I expect we will see reductions in the hundreds of nanoseconds in the execution times. :smiley:

**Comment by dmbates** [comment]
@palday In my local copy of the repository I did a rebase on current master, which worked without a hitch.  Is it necessary to do a rebase and force push or does the squash and merge take care of that because there is no overlap?

**Comment by palday** [comment]
`squash` should remove any potential redundancies because collapses things down into final differences. You may want to edit the commit message to remove the notes from intermediate commits. Force-pushing at this point wouldn't hurt, but I don't think it's necessary.

**Pull request merged**